### PR TITLE
fix: ensure reserved keywords can never be rendered for JS output

### LINKED ISSUE #176: language specific keywords are rendered as property names
#### Describe the bug
There are certain keywords that are reserved. This is different between languages, we need to make sure we dont render such properties as-is.

#### How to Reproduce
For example, in Java, you cannot have a property named `enum`. 

#### Expected behavior
I expected the library to handle any property names.

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR ensures that reserved keywords cannot be rendered for JS class as well as restricting TS outputs for the same keywords.

**Related issue(s)**
fixes #176

### CODE REVIEW

File: src/generators/javascript/JavaScriptRenderer.ts [hunk 1]
```diff
@@ -3,7 +3,201 @@ import { JavaScriptGenerator, JavaScriptOptions } from './JavaScriptGenerator';
...
+  'setTimeout',
+  'status', 
+  'submit', 
+  'taint', 
+  'text',
+  'textarea', 
+  'top', 
+  'unescape', 
+  'untaint',
+  'window'
```
Comment (magicmatatjahu) [inline src/generators/javascript/JavaScriptRenderer.ts:199]: These same we should use also in TS. Also please move the all arrays (also for other langs) to the separate file like `constants.ts`:
- generators/typescript/constant.ts
- generators/java/constant.ts

etc.

File: test/generators/javascript/JavaScriptGenerator.spec.ts [hunk 2]
```diff
@@ -6,6 +6,41 @@ describe('JavaScriptGenerator', () => {
...
+        reservedEnum: { type: 'string' }
+      },
+      additionalProperties: false
+    };
+    const expected = `class Address {
+  reservedReservedEnum;
+  reservedEnum;
+
+  constructor(input) {
+    this.reservedReservedEnum = input.reservedReservedEnum;
```
Comment (magicmatatjahu) [inline test/generators/javascript/JavaScriptGenerator.spec.ts:24]: `enum` isn't keyword in JS, also not `interface` or `type` :)

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: test/generators/javascript/JavaScriptGenerator.spec.ts (hunk 2 above)
Comment (jonaslagoni) [reply test/generators/javascript/JavaScriptGenerator.spec.ts:24]: Based on https://www.w3schools.com/js/js_reserved.asp they are?

File: src/generators/javascript/JavaScriptRenderer.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/generators/javascript/JavaScriptRenderer.ts:199]: > These same we should use also in TS.
We should? Thought the transpile process would ensure compatibility there?

File: test/generators/javascript/JavaScriptGenerator.spec.ts (hunk 2 above)
Comment (magicmatatjahu) [reply test/generators/javascript/JavaScriptGenerator.spec.ts:24]: LOL https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#future_reserved_keywords I don't know about it. Ok, leave it :)

File: src/generators/javascript/JavaScriptRenderer.ts (hunk 1 above)
Comment (magicmatatjahu) [reply src/generators/javascript/JavaScriptRenderer.ts:199]: Nope, for example TS `window` will be `window` in transpiled JS, so we should also treat the below keywords as reserved for TS.

Comment (jonaslagoni) [reply src/generators/javascript/JavaScriptRenderer.ts]: Huh, fair enough 😄 Changed

**Review by magicmatatjahu — APPROVED** [review]
LGTM

**Pull request merged**